machine/machine_core/ssh_connection.py

```diff
@@ -30,6 +30,16 @@
 from . import timeout as timeoutlib
 
 
+def write_all(fd, data):
+    while len(data) > 0:
+        try:
+            written = os.write(fd, data)
+            data = data[written:]
+        except BlockingIOError:
+            time.sleep(0.1)
```

Comment:
:frowning_face:

Reply:
Sure, it's not nice -- this is a quickfix to unbreak stuff. I am not against rewiring this entire messy function, it's just not something that I want to do under pressure.